packages/sass-render/bin/sass-render.js

```diff
@@ -51,9 +52,28 @@ if (!source) {
 
 glob(source, (err, files) => {
   files
-    .filter(file => !path.basename(file).startsWith('_'))
-    .forEach(file => {
-      sassRender(file).catch(error => {
+    // .filter(file => !path.basename(file).startsWith('_'))
+    .map((file) => {
+      // if not partial, process like regular file
+      const isPartial = path.basename(file).startsWith('_');
```

Comment:
Avoid single-use variable noise, just put the condition directly into `if`.

Reply:
Often just using those to write more `transparent` code, so that you just glimpse on that, and know what it's doing. You prefer using comments for that?

Reply:
It's about finding the team-specific optimum. In short functions such as we have here, things are clearly readable from code anyway.

Comments can help, but comment is only necessary when it actually has something to say differently than reading the very same thing as `if` statement already reads. Otherwise it just falls into DRY and now you have 2 things to keep up to date whenever you change something.

Reply:
Sounds like one of the multiple lectures of Robert C. Martin, like him I prefer to avoid comments whenever possible, 
yet something like this
```
if (!path.basename(file).startsWith('_') ) {
        return file;
}
```
Does not really tell me why is this checking '_' at the first look, does it?

Reply:
For such `if`, comment like "Partial?" makes perfect sense, since that term is repeated nowhere in the machine clause.

That's the main criteria - are we repeating words?

I've seen pretty dumbass stuff like

```php
// If thing doesn't exist
if ( ! $thing )
```

That just :facepalm:

Reply:
Also, find ways to deliver most meaning with least words.

Lazy `// Is it a partial?`

Optimized `// Partial?`

All sorts of articles, and other human language helpers often create noise that take away attention from the actual meat - you need to read and find the meat. Make finding the meat as simple as possible, where the other end limit is "avoid overly cryptic".

Reply:
comments fixed